enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,244 @@
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kubelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet. We work closely with the
+  containerd open source community and try to resolve it as soon as possible. We aim to fix any critical issue in the
+  upstream first and then update the containerd submodule to point to the latest containerd upstream version.
+  As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release. We will use two test jobs to test all e2e test cases
+  in the 4.10 community branch(OKD) and 4.10 branch(ccm OCP)to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
+  gets merged, if Windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. To overcome this issue, the user should log in to Windows VM and install `ctr or crictl`
+  command-line tool. Users can use one of these tools to pull the container image manually. Upon successful image download,
+  the user can create pods. Customers also can bake required container images into Windows VM image.
+* Currently Windows_exporter has been used to collect metrics from Windows node. We do
+  have containerd support in https://github.com/prometheus-community/Windows_exporter/releases/tag/v0.16.0
+  As of now, there is no feature parity mismatch between Docker and containerd.
+
+#### Removing a deprecated feature
+Once containerd becomes the default runtime, Docker will no longer be needed in Kubernetes stack. This is discussed in
+design section.
+
+### API Extensions
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+No API. N/A
+
+#### Support Procedures
+
+If containerd service doesn't get started, kubelet will not be in ready state. Windows node will not be able to schedule
+any pods. Containerd logs should be collected via must-gather script. We will check containerd service status and update the
+WMCO event log if the service is not up and running.
+
+### Test Plan
+
+We will use two test jobs OKD and existing ccm OCP to test the WMCO community operator. We will pass the WMCO CLI option
```

Comment:
Why do we need two OKD test jobs? I thought you would modify the existing OKD test job to use the new WMCO CLI option to pick containerd?

Reply:
two jobs ( one for OKD and one for OCP ). will rewrite it.